Tests/Runtime/CmColliderTests.cs

```diff
@@ -0,0 +1,160 @@
+
+            var obstacle = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            obstacle.transform.position = originalCamPosition; // place obstacle so that camera needs to move
+
+            yield return IncreaseTimeAndWaitForOnePhysicsFrame(m_DeltaTime);
+            m_Brain.ManualUpdate();
+            
+            // we are pulling away from obstacle
+            Assert.That(originalCamPosition, !Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+            var previousPosition = m_Vcam.State.FinalPosition;
+
+            CinemachineCore.CurrentTimeOverride += 0.5f;
+            yield return null;
+            m_Brain.ManualUpdate();
+            
+            Assert.That(previousPosition, !Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+            
+            UnityEngine.Object.Destroy(obstacle);
+
+            yield return IncreaseTimeAndWaitForOnePhysicsFrame(m_DeltaTime);
+            m_Brain.ManualUpdate();
+            
+            Assert.That(originalCamPosition, Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+        }
+        
+        [UnityTest]
+        public IEnumerator CheckDamping()
+        {
+            CinemachineCore.CurrentTimeOverride = 0;
+            m_Collider.m_SmoothingTime = 0;
+            m_Collider.m_Damping = 1;
+            m_Collider.m_DampingWhenOccluded = 0;
+            var originalCamPosition = m_Vcam.State.FinalPosition;
+
+            CinemachineCore.CurrentTimeOverride += m_DeltaTime;
+            yield return null; 
+            m_Brain.ManualUpdate();
+            
+            Assert.That(originalCamPosition, Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+
+            var obstacle = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            obstacle.transform.position = originalCamPosition; // place obstacle so that camera needs to move
+
+            yield return IncreaseTimeAndWaitForOnePhysicsFrame(m_DeltaTime);
+            m_Brain.ManualUpdate();
+            
+            // camera moved check
+            Assert.That(originalCamPosition, !Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+            
+            UnityEngine.Object.Destroy(obstacle);
+            var previousPosition = m_Vcam.State.FinalPosition;
+            
+            yield return IncreaseTimeAndWaitForOnePhysicsFrame(m_DeltaTime);
+            m_Brain.ManualUpdate();
+            
+            Assert.That(previousPosition, !Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance));
+            
+            // Because of frame delta in CinemachineCore, this cannot be checked!
+            // Assert.That(originalCamPosition, !Is.EqualTo(m_Vcam.State.FinalPosition).Using(Vector3EqualityComparer.Instance)); 
```

Comment:
A minimal hack fix would be to add an internal bool to CinemachineCore to turn off frame delta compensation

Reply:
That would work. To have an option to turn it off. Then our tests will be stable, and we can have deltaTimeOverride = 0.1 working deterministicly.